test/unit/lib/runner/writers/action-json-writer-tests.js

```diff
@@ -0,0 +1,798 @@
+          },
+        ],
+      };
+
+      passingReport = {
+        testReports: [
+          {
+            status: 'fail',
+            testRuns: [
+              {
+                report: {
+                  testName: 'testOne',
+                  actions: [
+                    {
+                      component: 'navigate',
+                      action: 'navigate',
+                      status: 'pass',
+                      time: [
+                        1,
+                        201697760,
+                      ],
+                      steps: {
+                        precondition: {
+                          status: 'pass',
+                          error: null,
+                        },
+                        perform: {
+                          status: 'pass',
+                          error: null,
+                        },
+                        effects: {
+                          status: 'pass',
+                          error: null,
+                        },
+                      },
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+        ],
+      };
+
+      mockery.enable({useCleanCache: true});
+      mockery.registerAllowable('../../../../../lib/runner/writers/action-json-writer.js');
+      mockery.registerMock('fs', fs);
+      mockery.registerMock('path', path);
+      mockery.registerMock('../../util/config-handler.js', configHandler);
+
+      actionJsonWriter = require('../../../../../lib/runner/writers/action-json-writer.js');
+      actionJsonWriter._getActionTestData = sinon.stub();
+      actionJsonWriter._writeReports = sinon.stub();
+      actionJsonWriter._checkActionForError = sinon.stub();
+    });
+
+    afterEach(function() {
+      mockery.resetCache();
+      mockery.deregisterAll();
+      mockery.disable();
+    });
+
+    it('should call configHandler.get with \'testPath\'', function() {
+      actionJsonWriter._getActionTestData.returns({
+        count: 0,
+      });
+
+      actionJsonWriter.write(report);
+
+      expect(configHandler.get.args).to.deep.equal([['testPath']]);
+    });
+
+    it('should call path.resolve with testPath', function() {
+      actionJsonWriter._getActionTestData.returns({
+        count: 0,
+      });
+      configHandler.get.returns('./testPath');
+
+      actionJsonWriter.write(report);
+
+      expect(path.resolve.args).to.deep.equal([['./testPath']]);
+    });
+
+    it('should call fs.readFileSync with the path to the test and \'utf8\'', function() {
+      path.resolve.returns('./testpath');
+      actionJsonWriter._getActionTestData.returns({
+        count: 0,
+      });
+
+      actionJsonWriter.write(report);
+
+      expect(fs.readFileSync.args).to.deep.equal([
+        [
+          './testpath/testOne',
+          'utf8',
+        ],
+      ]);
+    });
+
+    it('should call actionJsonWriter._getActionTestData with action name passed in', function() {
```

Comment:
this needs to be nested again with the next for loop `let action of run.report.actions`